File: typescript/src/renderer.ts
import { Shape } from './shapes/shape.js'

interface Renderer {
  draw: () => void
}

function newRenderer (figure: Shape): Renderer {
  return {
    draw () {
      const area: number = figure.computeArea()
      console.log('Shape drawn\n' + 'Its area is ' + area)
    }
  }
}

export { Renderer, newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer {
    private Shape figure;
    
    Renderer(Shape figure) {
        this.figure = figure;
    }

    void draw() {
        double area = figure.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
